src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -1158,8 +1158,6 @@ private void ScanBlank()
         /// <summary>Scans \-style backreferences and character escapes</summary>
         private RegexNode? ScanBasicBackslash(bool scanOnly)
         {
-            Debug.Assert(_pos < _pattern.Length, "The current reading position must not be at the end of the pattern");
```

Comment:
removed these asserts because there'll be a null ref a couple lines down in each case; we don't put asserts in front of every possible null ref.

Reply:
Not sure I understood the comment, but it looks to me that these are asserting `The current reading position` not a null ref.

Reply:
I should have said IndexOutOfRangeException

Reply:
> we don't put asserts in front of every possible null ref.

No, but we do put asserts in places that are meant to act as preconditions / contracts.  These asserts make it really easy for a maintainer to see that these methods should only be called when there's pattern remaining.  I don't think removing them was valuable.